subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c

```diff
@@ -390,9 +397,13 @@ FUNC_NORETURN static void ec_host_cmd_thread(void *hc_handle, void *arg2, void *
 		.reserved = NULL,
 	};
 
+	__ASSERT(hc->state != EC_HOST_CMD_STATE_DISABLED, "HC backend not initialized");
+
 	while (1) {
+		hc->state = EC_HOST_CMD_STATE_RECEIVING;
 		/* Wait until RX messages is received on host interface */
 		k_sem_take(&hc->rx_ready, K_FOREVER);
+		hc->state = EC_HOST_CMD_STATE_PROCESSING;
```

Comment:
Is it redundant to set the state to processing here and in `ec_host_cmd_rx_notify`?

Reply:
Good point.